### DISCUSSION TITLE
V18 Custom-created component theming

### DISCUSSION THREAD

**Discussion by biziosan** [primefaces/community#2952] [PrimeNG]
I love the new theming support in V18! I would like to use it also for any custom Angular component I might need for my application. I was wondering if there was a way to use `definePreset` to define the theme for custom-defined components in Angular. If not, do you have any suggestions about how to use the new V18 theme framework to do that?

**Comment by qwadrox** [comment]
@biziosan Hi!
There is a whole section on this in the docs: https://v18.primeng.org/theming
I don't understand exactly what you mean by "custom-defined components," but I think you will find the answer in the docs. You can create your own theme per component theme, etc. 
You can also follow the creator on YouTube where he regularly posts guides: 
https://www.youtube.com/watch?v=J3KFw5sih98

Although this tutorial is for PrimeVue, the same applies to PrimeNG except for Pass through which hasn't been implemented yet.

**Comment by biziosan** [comment]
Hi there!

Thank you for taking the time to answer my question. I'll explain to you as I realize that I need to clarify. We are working on a large application that uses PrimeNG as its primary UI framework. I am using V18 because of the fantastic new theme management option (again, thank you!). However, I would also like to use it for components we specifically designed for the application, not just the PrimeNG components. 

Let me give you an example: imagine I have a component called `TopBar`. Its primary function is to show the application's logo. I can change the theme of the PrimeNG components, but I need to use another mechanism to style `TopBar` (say, change the background color to match the user's theme selection). 

With an extendable theme manager from PrimeNG, I could include the new styles for `TopBar` directly inside the themes:

```typescript
const Noir = definePreset(Aura, {
    components: {
        topbar: {
            root: {
                background: 'white'
            }
        }
    },
});

this.config.theme.set({ preset: Noir })
```

This new function would be greatly appreciated!

**Reply by samdeanpinknews** [reply]
Isn't this in the extend section of the theming preset? https://primeng.org/theming#extend

**Reply by aboudard** [reply]
> Isn't this in the extend section of the theming preset? https://primeng.org/theming#extend

Yeah, I was about to say the same, it does fit the question I guess :)
